fieldservice_route_vehicle/models/fsm_route_dayroute.py

```diff
@@ -8,10 +8,15 @@ class FSMRouteDayRoute(models.Model):
 
     @api.model
     def _get_default_vehicle(self, vals=None):
-        route = self.route_id
-        if not route and vals.get('route_id', False):
+        vehicle_id = False
+        if self.person_id:
+            vehicle_id = self.person_id.vehicle_id.id
```

Comment:
If we have a Person but it doesn't have a Vehicle, we won't try getting on from the route.
Suggestion:
```
        vehicle_id = (
            self.route_id.fsm_vehicle_id.id or
            self.route_id.fsm_vehicle_id.id)
```